firefox_puppeteer/api/places.py

```diff
@@ -89,13 +89,15 @@ def restore_default_bookmarks(self):
               let observer = {
                 observe: function (aSubject, aTopic, aData) {
                   Services.obs.removeObserver(observer, "bookmarks-restore-success");
+                  Services.obs.removeObserver(observer, "bookmarks-restore-failed");
 
-                  marionetteScriptFinished(true);
+                  marionetteScriptFinished(aTopic == "bookmarks-restore-success");
                 }
               };
 
               // Trigger the import of the default bookmarks
               Services.obs.addObserver(observer, "bookmarks-restore-success", false);
+              Services.obs.addObserver(observer, "bookmarks-restore-failed", false);
               BookmarkHTMLUtils.importFromURL(defaultBookmarks, true);
             """, script_timeout=10000)
         except TimeoutException:
```

Comment:
@whimboo Sorry for the late. I had some personal problems last month, but everything is pretty fine right now.
To be honest, I'm not sure if I understood correctly what I need to do, so I upload an image showing what I understood about it 

![firefoui](https://cloud.githubusercontent.com/assets/81703/6780847/809a8aa6-d160-11e4-96f1-659ff4976576.png)